phitonthel/cfi-bei-client
# Implement Peer Nomination Supervisor Nomination Based on User Level and

**Description:**

This pull request introduces significant improvements to the NominatePeersModal component, focusing on both peer and supervisor nominations. The peers are now correctly identified based on the user level, ensuring a more accurate and relevant selection. Additionally, the functionality for nominating a Supervisor has been integrated, providing a comprehensive nomination process.

**Changes Made:**

**1. Peer Nomination Logic:**
Modified the peer fetching logic to list users of the same level as the authenticated user (authUser), excluding authUser themselves.
Ensured a consistent user experience by sorting the fetched users.

**2. Supervisor Nomination Integration:**
Implemented functionality to nominate a Supervisor.
Added a dropdown for Supervisor selection, pre-populated based on the user's level and division.

**3. Code Quality and Comments:**
Added comments to elucidate the logic behind peer and supervisor fetching.
Included console logs for debugging and verification of the peer and supervisor candidates.

**4. UI Enhancements:**
Ensured a clean and user-friendly interface for the nomination process.
Added labels and structured the dropdowns for a seamless user experience.

**How to Test:**

_Peer Nomination:_
Open the Peer Nomination modal.
Verify that the "Peers" dropdown lists users at the same level as authUser, excluding authUser.
Check the console for logged peer candidates and ensure their correctness.

_Supervisor Nomination:_
In the same modal, check the "Supervisor" dropdown.
Ensure that it is pre-populated based on the user's level and division.
Select a Supervisor and proceed with the nomination to ensure functionality.
